### DISCUSSION TITLE
Using something more simple than capnp

### DISCUSSION THREAD

**Discussion by 0x241F31** [crazyscot/qcp#11] [General]
I tried to install program from cargo and saw that tool uses capnp (that's because I got an error that it's not installed).
What about replacing it with something smaller, like rkyv or bincode? I think capnp is more useful when you're trying to setup communication between different platforms, but this tool has both client and server in single language.

**Comment by crazyscot** [comment]
Thanks for the feedback. You make a very good point; I chose capnp purely because I had used it in the past. This would be a good enhancement and it could reduce complexity.

In the meantime I've added a note to the README that you need to have capnpc installed in order to build.

**Reply by 0x241F31** [reply]
@crazyscot https://david.kolo.ski/rust_serialization_benchmark this could be helpful

**Reply by crazyscot** [reply]
Thanks. This work is now underway, using `serde_bare`.
I foresee making further breaking changes to make to the protocol soon, so I may try to include them all in a single breaking release.

**Comment by crazyscot** [comment]
I just released v0.3.0, which removes the capnp dependency. Instead, we use [BARE](https://www.ietf.org/archive/id/draft-devault-bare-11.html) encoding.
